spec/controllers/no_businesses_controller_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe NoBusinessesController, type: :controller do
+  include ApiHelpers
+
+  render_views
+
+  let(:user) { FactoryBot.create(:user) }
+  let(:task_id) { '2d98639e-5260-411f-a5ee-61847a2e067c' }
+  let(:submission_id) { '9a5ef62c-0781-4f80-8850-5793652b6b40' }
+
+  before do
+    allow_any_instance_of(SubmissionsController).to receive(:current_user).and_return(user)
+    mock_task_with_framework_endpoint!
+    mock_no_business_endpoint!
+  end
+
+  describe 'GET #new' do
+    it 'asks the user to confirm they want to report no business' do
+      get :new, params: { task_id: task_id }
+      expect(response).to be_successful
+      expect(response.body).to have_content 'Report no business'
+    end
+  end
+
+  describe 'POST #create' do
+    it 'reports no business to the API and redirects the user to the resulting submission' do
+      post :create, params: { task_id: task_id }
+
+      expect(response)
+        .to redirect_to(task_submission_path(task_id: '2d98639e-5260-411f-a5ee-61847a2e067c', id: submission_id))
```

Comment:
I found it difficult to understand how the `submission_id` ended up being set to the value in the `let` block. I assume it's hiding in the `mock_no_business_endpoint!` stub?

Reply:
yeah, there are a quite a few "magic" ID strings floating around the specs that basically map to the ids from the mock responses, which isn't great because of both the duplication and the fact that it's not immediately clear where the value comes from, or even if it's correct. I think at some point soon we'll need to figure out a more stable, reliable and sustainable way to mock the API responses than what we have at the moment. I think constant-ising the IDs by specifying them with `let`s was my attempt to DRY things up a bit in the meantime, but clearly I didn't do a very good job because I miss the `task_id` on this line!

Reply:
@tekin Perhaps add a comment to explain where the `submission_id` is magically appearing from? I agree that a more flexible way of mocking the API is the way to go.